lib/game.ex

```diff
@@ -124,4 +128,128 @@ defmodule Game do
+
+        # If move is in available moves continue
+        if move in game.available_moves do
+          # IO.puts("BEST SCORE BEFORE ==== #{best_score}")
+          IO.puts("DOING == #{move} ")
+          # play the next available move
+          game = handle_move(move, game)
+          game = check_for_win(game)
+          {move, score(game)}
+
+        else
+          # If move is not in available moves skip
+          IO.puts("SKIPPING == #{move}\n")
+        end
+      end)
+
+    move_list = Enum.filter(score_move_list, fn elem -> elem != :ok end)
+    list_of_scores = Enum.map(move_list, fn {_move, score} -> score end)
+    max_score = Enum.max(list_of_scores)
+    index_of_max_score = Enum.find_index(move_list, fn {_move, score} -> score == max_score end)
+    {:ok, {best_move, _score}} = Enum.fetch(move_list, index_of_max_score)
+    best_move
+
+  end
+
+  # def minimax(game) do
+
+  # end
+
+  # def minimax(game) do
+  #   # return score if game is complete
+  #   score(game)
+
+  #   if (game[:maximizer] == game.current_player) do
+  #     # best_score = -100
+
+  #     move_score_map =
+  #       Enum.map([1, 2, 3, 4, 5, 6, 7, 8, 9], fn move ->
+  #         if move in get_available_moves(game) do
+  #           IO.puts("AVAIL MOVES BELOW")
+  #           IO.inspect(game.available_moves)
+  #           IO.puts("AVAIL MOVES ABOVE")
+
+  #           game = handle_move(move, game)
+  #           game = check_for_win(game)
+  #           # run the minimax algo and grab the score
+  #           score = minimax(game)
+  #           # if the score is larger than the best score return that move
+  #           # best_score = Enum.max(best_score, score)
+  #           {move, score}
+  #         end
+  #       end)
+
+  #     move_score_map
+  #   else
+  #     # best_score = 100
+
+  #     move_score_map =
+  #       Enum.map([1, 2, 3, 4, 5, 6, 7, 8, 9], fn move ->
+  #         if move in get_available_moves(game) do
+  #           IO.puts("AVAIL MOVES BELOW")
+  #           IO.inspect(game.available_moves)
+  #           IO.puts("AVAIL MOVES ABOVE")
+
+  #           game = handle_move(move, game)
+  #           game = check_for_win(game)
+  #           # run the minimax algo and grab the score
+  #           score = minimax(game)
+  #           # if the score is larger than the best score return that move
+  #           # best_score = Enum.min(best_score, score)
+  #           {move, score}
+  #         end
+  #       end)
+
+  #     move_score_map
+  #   end
+  # end
+
+  def assign_maximizer(game), do: %{game | maximizer: game.current_player}
+
+  def assign_winner(game), do: %{game | winner: game.current_player}
+
+  def score(%{win?: false, available_moves: []}), do: 0
+
+  def score(%{winner: winner, win?: true, maximizer: maximizer}) when winner == maximizer do
+    IO.puts("WINNER!")
+    10
+  end
+
+  def score(%{win?: true}) do
+    IO.puts("LOSER!")
+    -10
+  end
+
+  def score(game), do: game
```

Comment:
is this a totally separate return type? the other `score` fns all return an int, no?

Reply:
My intent was to have the game fall through the functions until it became an integer. Once it becomes an integer then we can grab it as the score. Still trying to work this one out.